References: bsc#1161480

Fix xl shutdown for HVM without PV drivers

A return value of zero means no PV drivers. Restore the hunk which was removed.

Fixes commit b183e180bce93037d3ef385a8c2338bbfb7f23d9

Signed-off-by: Olaf Hering <olaf@aepfle.de>
---
 tools/libxl/libxl_domain.c | 3 +++
 1 file changed, 3 insertions(+)

Index: xen-4.13.1-testing/tools/libxl/libxl_domain.c
===================================================================
--- xen-4.13.1-testing.orig/tools/libxl/libxl_domain.c
+++ xen-4.13.1-testing/tools/libxl/libxl_domain.c
@@ -795,6 +795,9 @@ int libxl__domain_pvcontrol(libxl__egc *
     if (rc < 0)
         return rc;
 
+    if (!rc)
+        return ERROR_NOPARAVIRT;
+
     shutdown_path = libxl__domain_pvcontrol_xspath(gc, domid);
     if (!shutdown_path)
         return ERROR_FAIL;
